advection_test/functions.py

```diff
@@ -0,0 +1,1302 @@
+    - title: set title of the plot
+
+    Outputs: none
+    """
+
+    fig = plt.figure(figsize=(8, 4))
+    fig.patch.set_facecolor("white")
+    ax = fig.add_subplot(111, projection=ccrs.Robinson())
+    ax.set_facecolor(".4")
+
+    f1 = pcolormesh_cube(lat, lon, field, cmap=cmap, vmin=vmin, vmax=vmax)
+    plt.colorbar(f1, label=units)
+
+    ax.set_title(title)
+
+    if fSave is not None:
+        plt.savefig(fSave, dpi=200, bbox_inches="tight")
+        plt.close("all")
+    else:
+        plt.show()
+
+    return
+
+
+def plot_grid(
+    lon, lat, field, cmap="bwr", vmin=-1, vmax=1, units="", title="", fSave=None
+):
+    """
+    Use: plot_projection_field(
+        lon, lat, field, cmap='viridis', vmin=-1, vmax=1, units='', title='')
+
+    Creates a Robinson projection and plots the (6) tiles on a map.
+
+    Inputs:
+    - lon, lat: lon and lat of coordinate edges (tile, x, y)
+    - field: unstaggered field at a given vertical level (tile, x, y)
+    - cmap: colormap
+    - vmin, vmax: limits of the color map
+    - units: label the units on the color bar
+    - title: set title of the plot
+
+    Outputs: none
+    """
+
+    fig = plt.figure(figsize=(8, 4))
+    fig.patch.set_facecolor("white")
+    ax = fig.add_subplot(111, projection=ccrs.Robinson())
+    ax.set_facecolor(".4")
+
+    f1 = pcolormesh_cube(
+        lat, lon, field, cmap=cmap, vmin=vmin, vmax=vmax, edgecolor="k", linewidth=0.1
+    )
+    # plt.colorbar(f1, label=units)
+
+    ax.set_title(title)
+
+    if fSave is not None:
+        plt.savefig(fSave, dpi=300, bbox_inches="tight")
+        plt.close("all")
+    else:
+        plt.show()
+
+    return
+
+
+def write_initialCondition_toFile(
+    fOut, variables, dimensions, units, origins, backend, configuration, mpi_rank
+):
+    """
+    Use: write_initialCondition_toFile(
+        fOut, lon, lat, data, dimensions, units, communicator)
+
+    Inputs:
+    - fOut: output netcdf file
+    - variables: Dict{'delp', 'smoke', 'uC', 'vC'}
+    - dimensions: Dict{'tile', 'nx', 'ny', 'nx1', 'ny1'}
+    - units: Dict{'coord', 'wind', 'pressure', 'mass'}
+    - origins: Dict{'compute_2d'}
+    - backend: 'numpy'
+    - configuration: Dict{'communicator', 'grid_data'}
+    - mpi_rank: gathered from all tiles, only written on root
+
+    Outputs: None; creates netCDF file
+    """
+
+    lon, lat = get_lonLat_edges(
+        configuration["grid_data"], dimensions, units, origins, backend
+    )
+
+    lon_global = configuration["communicator"].gather(lon)
+    lat_global = configuration["communicator"].gather(lat)
+
+    uC = np.squeeze(configuration["communicator"].gather(variables["uC"]))
+    vC = np.squeeze(configuration["communicator"].gather(variables["vC"]))
+    delp = np.squeeze(configuration["communicator"].gather(variables["delp"]))
+    smoke = np.squeeze(configuration["communicator"].gather(variables["smoke"]))
+
+    if mpi_rank == 0:
+        uC = unstagger_coordinate(uC)
```

Comment:
why do we unstagger in place?

Reply:
It's only inside the function that writes the data -- the plotting function only wants data on cell centers, so when you store it, it's easier to just do it with unstaggered data?